pkg/ebpf/dns_cache.go

```diff
@@ -17,25 +17,32 @@ type reverseDNSCache struct {
 	// that this file can run on a 32 bit system, they must 64-bit aligned.
 	// Go will ensure that each struct is 64-bit aligned, so these fields
 	// must always be at the beginning of the struct.
-	length   int64
-	lookups  int64
-	resolved int64
-	added    int64
-	expired  int64
+	length    int64
+	lookups   int64
+	resolved  int64
+	added     int64
+	expired   int64
+	oversized int64
 
 	mux  sync.Mutex
 	data map[util.Address]*dnsCacheVal
 	exit chan struct{}
 	ttl  time.Duration
 	size int
+
+	// maxDomainsPerIP is the maximum number of domains mapped to a single IP
+	maxDomainsPerIP   int
+	oversizedLogLimit *util.LogLimit
 }
 
 func newReverseDNSCache(size int, ttl, expirationPeriod time.Duration) *reverseDNSCache {
 	cache := &reverseDNSCache{
-		data: make(map[util.Address]*dnsCacheVal),
-		exit: make(chan struct{}),
-		ttl:  ttl,
-		size: size,
+		data:              make(map[util.Address]*dnsCacheVal),
+		exit:              make(chan struct{}),
+		ttl:               ttl,
+		size:              size,
+		oversizedLogLimit: util.NewLogLimit(10, time.Minute*10),
+		maxDomainsPerIP:   1000,
```

Comment:
Do we want to make it a configurable value?

Reply:
Nothing else here is configurable, and very few installations should hit this 1000 limit.